proto/frequenz/api/reporting/v1/reporting.proto

```diff
@@ -0,0 +1,534 @@
+  //        Example: `sum(#1,#2)`, `avg(#3,#4)`
+  //
+  //     2. Dedicated formulas: These support basic math operators while concatenating microgrid
+  //        component IDs.
+  //        Example: `#1 + #2 - #3`, `(#3 * #2) /# 1`
+  string aggregation_formula = 3;
+}
+
+// Defines a single aggregated metric sample, which represents a consolidated metric value
+// over a collection of microgrid components within a specific timeframe.
+//
+// This message serves to encapsulate aggregate metrics derived from multiple component
+// measurements. It is particularly useful in scenarios where an overall metric needs to be
+// calculated for a set of components. For instance, you could use it to represent the average
+// voltage across multiple sources within a microgrid or to calculate the average state of
+// charge of several batteries.
+//
+message SimpleAggregatedMetricSample {
+  // The UTC sample timestamp for the aggregated metrics.
+  //
+  // !!! info
+  //     This timestamp indicates the specific moment when the aggregated metric value
+  //      is calculated.
+  google.protobuf.Timestamp sampled_at = 1;
+
+  // The aggregated value of the metric.
+  // float value = 2;
+  frequenz.api.common.v1.metrics.SimpleMetricSample sample = 2;
+}
+
+// Message defining the request format for fetching historical metrics, such as electrical
+// measurements, and other information for individual microgrid components.
+//
+// !!! note
+//     In addition to the raw metrics, the API can also return additional information
+//     like errors or operational states of the components during the specified time period.
+//
+message ListMicrogridComponentsDataRequest {
+  // General filter criteria for querying microgrid components data.
+  message ListFilter {
+    // Optional resampling options like resolution for the data, represented in seconds.
+    // If omitted, data will be returned in its original representation.
+    ResamplingOptions resampling_options = 1;
+
+    // Optional time-based filter criteria.
+    TimeFilter time_filter = 2;
+
+    // Include options specifying additional fields to be included in the response.
+    IncludeOptions include_options = 3;
+  }
+
+  // Encapsulates the microgrid ID and the component IDs within that microgrid for which
+  // the historical data should be retrieved.
+  //
+  // !!! note
+  //     Each entry in this repeated field associates a microgrid ID with its respective
+  //     component IDs. At least one such association must be provided for a valid request.
+  repeated frequenz.api.common.v1.microgrid.MicrogridComponentIDs microgrid_components = 1;
+
+  // List of metrics to return. Only the specified metrics will be returned.
+  //
+  // !!! note
+  //     At least one metric must be specified. Failure to do so will result in an empty response.
+  repeated frequenz.api.common.v1.metrics.Metric metrics = 2;
```

Comment:
>For instance it would not be possible to query battery SOC and meter power with a single request.

Interesting. Why do we need this restriction?

Reply:
We don't need such a restriction, but IIUC you cannot request the following:
```
metrics = [SOC, power]
microgrid_components = [ {MID, CID_meter}, {MID, CID_battery} ]
```
since it creates the cross-combinations of these, i.e. 4 values. But for CID_meter you don't have SOC.

That would be solved if the metric is combined with the components.

Reply:
Oh now I see what you mean. Yeah, makes sense.
I'd suggest creating an issue for it.

Reply:
Don't we want to fix this in the first version?

Reply:
This is an interesting conundrum. Should we create a message type that combines `MicrogridComponentIDs` and `Metrics`? Should there be a list of valid metrics for a given component type maintained somewhere? 

I think this will require some effort to solve, so unless it completely breaks the use case, perhaps we can solve it in a later version of the API?

Reply:
>Don't we want to fix this in the first version?

No, I'd not suggest that.

Reply:
Issue created.